components/places/src/types.rs

```diff
@@ -102,6 +101,22 @@ impl<'de> serde::Deserialize<'de> for VisitTransition {
     }
 }
 
+#[derive(Clone, PartialEq)]
```

Comment:
Apart from using a `Url` where possible, I think I'd rather see these new types in ffi.rs - IIUC, they exist only for the ffi and aren't used by the actual implementation, but putting them here makes them look like "core" structs.